agent/xds/resources_test.go

```diff
@@ -314,7 +316,13 @@ AAJAMaoXmoYVdgXV+CPuBb2M4XCpuzLu3bcA2PXm5ipSyIgntMKwXV7r
 -----END CERTIFICATE-----`
 )
 
-func getAPIGatewayGoldenTestCases() []goldenTestCase {
+func getAPIGatewayGoldenTestCases(t *testing.T) []goldenTestCase {
+	t.Helper()
+
+	service := structs.NewServiceName("service", nil)
+	serviceUID := proxycfg.NewUpstreamIDFromServiceName(service)
+	serviceChain := discoverychain.TestCompileConfigEntries(t, "service", "default", "default", "dc1", connect.TestClusterID+".consul", nil)
```

Comment:
The argument list for this feels unfortunately confusing and could be more clear as fields on a single struct arg, but assuming that's out of scope for this change.

Reply:
Yeah, this is primarily just copying from what other tests are doing -- many of these xDS golden file generation tests are like this and are fairly opaque until you've poked around a good bit.